### PR TITLE
docs: concepts updates for v0.9

### PR DESCRIPTION
**Pull request by killme2008**
## What's Changed in this PR

*Describe the change in this PR*

Update concepts.

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Updated architecture documentation to clarify roles and interactions of GreptimeDB components.
  - Enhanced data model documentation with detailed explanations of table columns and indexing.
  - Refined key concepts documentation to better describe database components and their functionalities.
  - Revised user guide overview to emphasize GreptimeDB's flexible architecture for metrics and event data.
  - Improved Chinese documentation with detailed descriptions of data models and key concepts.

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
The ZH docs should be updated also?

**Comment by nicecui** [comment]
I'll add the zh docs right now

**Comment by killme2008** [comment]
Please don't merge it, I'll modify it tomorrow if I feel better.

**Comment by killme2008** [comment]
@nicecui Already done. Please take a look. If everything is good, we can copy them into `v0.9` folder too.

File: docs/nightly/en/user-guide/concepts/why-greptimedb.md [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+GreptimeDB is a cloud-native, distributed and open source time series database, it's designed to process, store and analyze vast amounts of metrics, logs & events data (also Traces in plan).
+
 It's highly efficient at handling hybrid processing workloads which involve both time-series and real-time analysis, while providing users with great experience.
-To gain insight into the motivations that led to the development of GreptimeDB, we recommend reading our blog post titled ["This Time, for Real"](https://greptime.com/blogs/2022-11-15-this-time-for-real).
-In this article, we delve into the reasons behind Greptime's high performance and some highlighted features.
+
+To gain insight into the motivations that led to the development of GreptimeDB, we recommend reading our blog posts titled ["This Time, for Real"](https://greptime.com/blogs/2022-11-15-this-time-for-real) and ["Unifying Logs and Metrics"](https://greptime.com/blogs/2024-06-25-logs-and-metrics).
+
+In this document, we delve into the reasons behind Greptime's high performance and some highlighted features.
```
Comment (nicecui) [inline docs/nightly/en/user-guide/concepts/why-greptimedb.md:9]:
```suggestion
In these documents, we delve into the reasons behind Greptime's high performance and some highlighted features.
```

File: docs/nightly/en/user-guide/concepts/why-greptimedb.md [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+
+## Unified metrics, logs and events
+
+Through the model design of [time series tables](./data-model), native support for SQL, and the hybrid workload brought by the storage-computation separation architecture, GreptimeDB can handle metrics, logs, and events together, enhance the coloration analysis between different time series data and simplify the architecture, deployment and APIs for users.
```
Comment (nicecui) [inline docs/nightly/en/user-guide/concepts/why-greptimedb.md:13]:
```suggestion
Through the model design of [time series tables](./data-model), native support for SQL, and the hybrid workload brought by the storage-computation separation architecture, GreptimeDB can handle metrics, logs, and events together, enhance the correlation analysis between different time series data and simplify the architecture, deployment and APIs for users.
```

**Review by nicecui — COMMENTED** [review]
LGTM, please add docs to v0.9

**Review by nicecui — APPROVED** [review]

**Pull request merged**